### ISSUE TITLE
rtc

### ISSUE THREAD

**Issue by DR-Soaps** [spiritualized1997/openFPGA-GB-GBC#8]
rtc is not working after first start

**Comment by kresp0** [comment]
Partial RTC support [have been added](https://www.analogue.co/developer/docs/changelog/1-1-beta-6) to the openFPGA 1.1 beta 6.

It would be nice if this core added support for RTC.

**Comment by Guzaboru** [comment]
Hopefully we will get the real time clock for gbc and gba sometime.

**Comment by dannenberg** [comment]
i too noticed this and would be excited to see it added/working

**Comment by zenzoa** [comment]
I would love this! I'm working on a virtual pet for the gameboy, and having RTC support would mean being able to have idle-game elements.

**Comment by LeeNeighoff** [comment]
+1 for RTC support now that it's available to openFPGA cores.

**Comment by DunhamGitHub** [comment]
> +1 for RTC support now that it's available to openFPGA cores.

Can you elaborate "now that it's available to openFPGA cores"? I have not seen any update from Analogue's side🧏🏼‍♀️

**Comment by kresp0** [comment]
> Can you elaborate "now that it's available to openFPGA cores"? I have not seen any update from Analogue's side🧏🏼‍♀️

Partial support was added on November 2022 to [openFPGA 1.1 beta 6](https://www.analogue.co/developer/docs/changelog/1-1-beta-6):

> Cores get notified of the current Pocket date and time when they start

**Comment by DunhamGitHub** [comment]
> > Can you elaborate "now that it's available to openFPGA cores"? I have not seen any update from Analogue's side🧏🏼‍♀️
> 
> Partial support was added on November 2022 to [openFPGA 1.1 beta 6](https://www.analogue.co/developer/docs/changelog/1-1-beta-6):
> 
> > Cores get notified of the current Pocket date and time when they start

Fair enough. May I kindly point out that "manually changing the in game clock" can simply be patched into the ROM (walk up to the clock inside every Pokecenter and just change the time)...
99.9% of games (other than Pokémon) don't use RTC so I can guess this is not a priority at all...

**Comment by ninbura** [comment]
Specifically the `GETTIME` method.
```
Added Chip32 instruction GETTIME
```

From my understanding agg added rtc to his [snes core](https://github.com/agg23/openfpga-SNES/releases/tag/0.4.0). 

If this does end up getting some development time, might I suggest trying to replicate the RTC functionality that EZ-Flash flash carts have. From what I can tell, upon launching a game; the difference between the last recorded timestamp and the current timestamp is calculated. After which new RTC data is injected to account for time passed since the game was last run. Ultimately, this allows for RTC to function in a natural way even with the use of save states and sleep (which is just save states in disguise).

Here's [an example](https://reddit.com/r/AnaloguePocket/s/vyThyAh18z) of this in action.
